### PR TITLE
Updated Readme / Added EggTimer Support

### PR DESCRIPTION
**Pull request by bshep**
* Added 'EggTimer' support, the function is called setCircuitRuntimeById
* Updated readme with information about the new functions

### CODE REVIEW

**Comment by parnic** [comment]
Hey @bshep, as soon as the review comments are addressed, I'm happy to merge this.

**Comment by mikemucc** [comment]
So glad to see more functionality getting added!

Commenting so that Github will let me know when this gets merged.

@parnic are you planning to do an npm deploy after this gets merged? When you do, we're going to start incorporating this into my API.

**Comment by parnic** [comment]
@mikemucc FYI you can subscribe to an issue without commenting :).

I can update the npm package whenever, sure.

**Comment by mikemucc** [comment]
Or course ;-)
I also wanted to show my support to @bshep, so cool to see more functionality getting added.

This could potentially become a full replacement for Pentair's apps. And as Pentair has decided not to support 64-bit mac (or mac at all truthfully, as their app is just the windows version running under Wine), the sooner the better.

...now, does anybody know anything about electron?

**Comment by bshep** [comment]
> Hey @bshep, as soon as the review comments are addressed, I'm happy to merge this.

Hmmm not sure which comments you mean? The CI checks seem to have passed? I think its an issues with the way i committed things to git, do you have any suggestions on how to fix it?

File: README.md [hunk 1]
```diff
@@ -476,3 +481,57 @@ Passed as an argument to the emitted `gatewayFound` event. Contains information
...
 * `portOpen` - boolean indicating whether or not the port is open and able to be connected to
 * `relayOn` - boolean indicating whether the relay is on (unsure what exactly this indicates; it's always been false in my tests)
+
+## SLAddNewScheduleEvent
+
+Adds a new event to the specified schedule type, either 0 for regular events or 1 for one-time events
+
+### Properties
+
+* `scheduleType` - 0 - indicates regular scheduled events, 1 - indicates
```
[the remark this answers is not in the document]
Comment (bshep) [reply README.md]: fixed in new commit

File: README.md [hunk 2]
```diff
@@ -476,3 +481,57 @@ Passed as an argument to the emitted `gatewayFound` event. Contains information
...
+
+* `scheduleId` - the scheduleId of the schedule to be deleted
+
+## SLGetScheduleData
+
+Retrieves a list of schedule events of the specified type, either 0 for regular events or 1 for one-time events
+
+### Properties
+
+* `scheduleType` - 0 - indicates regular scheduled events, 1 - indicates
```
[the remark this answers is not in the document]
Comment (bshep) [reply README.md]: fixed in new commit

File: README.md [hunk 3]
```diff
@@ -476,3 +481,57 @@ Passed as an argument to the emitted `gatewayFound` event. Contains information
...
+* `circuitId` - id of the circuit to which this event applies to
+* `startTime` - the start time of the event, specified as minutes since midnight
+	* example: 6:00am would be 360
+	* example: 6:15am would be 375
+* `stopTime` - the stop time of the event, specified as minutes since midnight, format is the same as startTime
+* `dayMask`
+	* 7-bit mask that determines which days the schedule is active for, MSB is always 0, valid numbers 1-127
+* `flags`
+	* bit 0 is the schedule type, if 0 then regular event, if 1 its a run-once
+	* bit 1 indicates wether heat setPoint should be changed
```
[the remark this answers is not in the document]
Comment (bshep) [reply README.md]: fixed in new commit

**Comment by bshep** [comment]
I think i fixed them... sorry about the missing explanations, copy-paste error ;-)

**Pull request merged**